### PR TITLE
[Release] 1.0.0-rc.1

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @rmgogogo 
Bumping main version by merging release branch back.

### CODE REVIEW

File: backend/api/python_http_client/kfp_server_api/__init__.py [hunk 1]
```diff
@@ -21,15 +21,15 @@
 
     This file contains REST API specification for Kubeflow Pipelines. The file is autogenerated from the swagger definition.  # noqa: E501
 
-    The version of the OpenAPI document: 1.0.0-dev.1
+    The version of the OpenAPI document: 1.0.0-rc.1
```
Comment (Ark-kun) [inline backend/api/python_http_client/kfp_server_api/__init__.py:24]: Can we prevent changing all those files every time?

File: .cloudbuild.yaml [hunk 2]
```diff
@@ -56,14 +56,14 @@ steps:
 - name: 'gcr.io/cloud-builders/docker'
   args: ['build', '-t', 'gcr.io/$PROJECT_ID/frontend:$COMMIT_SHA',
          '--build-arg', 'COMMIT_HASH=$COMMIT_SHA',
-         '--build-arg', 'TAG_NAME=$TAG_NAME',
+         '--build-arg', 'TAG_NAME=$(cat /workspace/VERSION)',
```
Comment (Ark-kun) [inline .cloudbuild.yaml:59]: Are you sue we're not overwriting the released tag on every master commit?

File: backend/api/python_http_client/kfp_server_api/__init__.py (hunk 1 above)
Comment (Bobgy) [reply backend/api/python_http_client/kfp_server_api/__init__.py:24]: If we decide to commit these into the repo, they will be automatically updated each release.

Other than that, I don't think that's possible. What concerns do you have?

File: .cloudbuild.yaml (hunk 2 above)
Comment (Bobgy) [reply .cloudbuild.yaml:59]: this is not up-to-date in fact

Comment (Bobgy) [reply .cloudbuild.yaml:59]: also it's kind of weird this shows up in the diff, this change was cherry picked from master.

File: components/gcp/dataproc/submit_sparksql_job/component.yaml [hunk 3]
```diff
@@ -75,7 +75,7 @@ outputs:
     type: UI metadata
 implementation:
   container:
-    image: gcr.io/ml-pipeline/ml-pipeline-gcp:ad9bd5648dd0453005225779f25d8cebebc7ca00
+    image: gcr.io/ml-pipeline/ml-pipeline-gcp:1.0.0-rc.1
```
Comment (Ark-kun) [inline components/gcp/dataproc/submit_sparksql_job/component.yaml:78]: There have been no changes to this image. Can we avoid changing the component and references?
We've got feedback from a customer that it's hard to track component updates and understand what has changed.

File: sdk/python/requirements.in [hunk 4]
```diff
@@ -10,7 +10,7 @@ jsonschema>=3.0.1
 kubernetes>=8.0.0, <12.0.0
 
 # kfp.Client
-kfp-server-api>=0.2.5, <0.4.0  #Update the upper version whenever a new version of the kfp-server-api package is released. Update the lower version when there is a breaking change in kfp-server-api, or kfp sdk depends on new api changes in kfp-server-api.
+kfp-server-api>=0.2.5, <2.0.0
```
Comment (Ark-kun) [inline sdk/python/requirements.in:13]: Do the update rules no longer apply?

Comment (Bobgy) [reply sdk/python/requirements.in:13]: I moved update rules to setup.py file and also mentioning to update this file.

File: components/gcp/dataproc/submit_sparksql_job/component.yaml (hunk 3 above)
Comment (Bobgy) [reply components/gcp/dataproc/submit_sparksql_job/component.yaml:78]: I've summarized my opinions on the email thread. Would you mind replying there so we have a centralized discussion? It's still very easy to change this to optional now.

**Comment by Bobgy** [comment]
Hmmm, actually I realized I shouldn't create a PR from release-1.0 branch, because I will have no way to resolve conflicts...

**Comment by Bobgy** [comment]
@Ark-kun I will create a new PR instead

**Comment by Bobgy** [comment]
Superceded by https://github.com/kubeflow/pipelines/pull/3927

**Pull request closed without merging**